src/mixins/filing-template-mixin.ts

```diff
@@ -30,58 +30,57 @@ export default class FilingTemplateMixin extends Vue {
   @Action setDefineCompanyStepValidity!: ActionBindingIF
   @Action setNameRequestState!: ActionBindingIF
   @Action setOrgPersonList!: ActionBindingIF
-  @Action setFilingId!: ActionBindingIF
   @Action setCertifyState!: ActionBindingIF
   @Action setShareClasses!: ActionBindingIF
   @Action setEffectiveDate!: ActionBindingIF
   @Action setIsFutureEffective!: ActionBindingIF
   @Action setFolioNumber!: ActionBindingIF
   @Action setIncorporationAgreementStepData!: ActionBindingIF
 
-  /** Constructs a filing body, used when saving a filing. */
+  /** Constructs a filing body from store data. Used when saving a filing. */
   buildFiling (): IncorporationFilingIF {
-    // Format DateTime for Filing
-    const effectiveDate = this.stateModel?.incorporationDateTime?.effectiveDate || ''
+    // Format DateTime for filing.
+    const effectiveDate = this.stateModel.incorporationDateTime.effectiveDate
     const formattedDateTime = effectiveDate &&
       (effectiveDate.toISOString()).replace('Z', '+00:00')
 
-    // Build and return filing
-    let filing:IncorporationFilingIF = {
+    // Build filing.
+    const filing: IncorporationFilingIF = {
       filing: {
         header: {
           name: INCORPORATION_APPLICATION,
-          certifiedBy: this.stateModel?.certifyState?.certifiedBy || '',
-          date: this.stateModel?.currentDate || '',
+          certifiedBy: this.stateModel.certifyState.certifiedBy,
+          date: this.stateModel.currentDate,
           folioNumber: this.stateModel.defineCompanyStep.folioNumber,
           isFutureEffective: this.stateModel.incorporationDateTime.isFutureEffective
         },
         business: {
-          legalType: this.stateModel?.entityType || '',
-          identifier: this.getTempId || ''
+          legalType: this.stateModel.entityType,
+          identifier: this.getTempId
         },
         incorporationApplication: {
-          offices: this.stateModel?.defineCompanyStep?.officeAddresses || {},
+          nameRequest: {
+            legalType: this.stateModel.entityType
+          },
+          offices: this.stateModel.defineCompanyStep.officeAddresses,
           contactPoint: {
-            email: this.stateModel?.defineCompanyStep?.businessContact?.email || '',
-            phone: this.stateModel?.defineCompanyStep?.businessContact?.phone || '',
-            extension: this.stateModel?.defineCompanyStep?.businessContact?.extension || ''
+            email: this.stateModel.defineCompanyStep.businessContact.email,
+            phone: this.stateModel.defineCompanyStep.businessContact.phone,
+            extension: this.stateModel.defineCompanyStep.businessContact.extension
           },
-          parties: this.stateModel?.addPeopleAndRoleStep?.orgPeople || [],
-          shareClasses: this.stateModel?.createShareStructureStep?.shareClasses || [],
+          parties: this.stateModel.addPeopleAndRoleStep.orgPeople,
```

Comment:
did we test this with an empty draft IA for numbered company? I know I was getting issues here because contact point/offices/parties aren't defined until at least a first save

Reply:
Yes, I created a new named IA and navigated to this UI and verified that the draft loaded OK and there were no errors.
In this method, a filing object is created from store data. That store data always exists (since that's how we declare the stateModel object). So, de-referencing its properties should always succeed.
I further confirmed that all stateModel properties/subobjects are already initialized in the store (to the same fallback values we had here in fact).
Do you remember how to reproduce the issues you saw?

Reply:
afaik there is zero validation on draft filings now. Was this an issue you encountered recently?

Reply:
No, this is just cleanup while I was already working on this method. But it did bug me that it looked like were providing useful fallbacks, but this code was unnecessary. I guess this principle is sort of DRY and sort of "anti decoy pattern" :)

Reply:
I do not remember...or at least I do remember a numbered company empty draft causing that issue. Before adding fallbacks I had used a method called "buildEmptyFiling()" for testing (if you remember). But, if you say it doesn't happen, I believe you!